src/Raven.Client/Documents/Changes/AggressiveCacheDatabaseChanges.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using Raven.Client.Http;
+
+namespace Raven.Client.Documents.Changes;
+
+internal class AggressiveCacheDatabaseChanges : DatabaseChanges
+{
+    internal AggressiveCacheDatabaseChanges(RequestExecutor requestExecutor, string databaseName) : base(requestExecutor, databaseName, onDispose: null, nodeTag: null)
```

Comment:
`OnDispose` should probably remove the connection from the dictionary. If it's faulted but still present in the dictionary then next time we try to use it we will get the faulted one instead of a new one

Reply:
done